python/pyspark/ml/linalg/__init__.py

```diff
@@ -705,6 +705,22 @@ def __eq__(self, other):
             return Vectors._equals(self.indices, self.values, list(xrange(len(other))), other.array)
         return False
 
+    def __getattr__(self, item):
+        def wrapper(*args, **kwargs):
+            if _have_scipy:
+                csr = scipy.sparse.csr_matrix((np.append(self.values, 0),
+                                               np.append(self.indices, self.size-1),
+                                               [0, len(self.values)]))
+                func = getattr(csr, item)
+                result = func(*args, **kwargs)
+                if isinstance(result, scipy.sparse.csr.csr_matrix):
+                    return SparseVector(result.shape[1],result.indices,result.data)
+                return result
+            else:
+                raise AttributeError(
+                    "'{0}' object has no attribute '{1}' or SciPy not installed.".format(self.__class__, item))
```

Comment:
There are 3 cases that can occur to call __getattr__ and get an error:
1) Calling a SciPy function while not having SciPy installed.
2) Calling a function that is not in SparseVector or SciPy, when SciPy is installed- SciPy provides the attribute error on its own.
3) Calling a function that is not in SparseVector or SciPy, without SciPy installed.

The message is accounting for cases 1 & 3. 
If User tries to call a function which does not exist in SparseVector or SciPy, while not having SciPy installed, they are warned that it might not exist at all as well as told they have to install SciPy as a possible solution.